### PR TITLE
Documentation updates

### PR DESCRIPTION
**Pull request by normandy7**
- Tweak phrasing and formatting
- Move `Returns` above `Example`
- Clarify ObjectState
- Document StringSet
- Mention column filtering limit

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -194,190 +209,224 @@ Representation of a Neptune run in a limited read-only mode.
...
 ```python
+run = ReadOnlyRun(...)
 run.prefetch(["parameters/optimizer", "parameter/init_lr"])
 # No more calls to the API
 print(run["parameters/optimizer"].fetch())
 print(run["parameter/init_lr"].fetch())
 ```
 
-## Available types
+## Supported field types
```
[the remark this answers is not in the document]
Comment (normandy7) [reply README.md:304]: Mentioning string sets was proposed to the internal draft on April 19 following https://github.com/neptune-ai/neptune-fetcher/pull/19/ but I see it's not included in the changelog 🤔

File: README.md [hunk 2]
```diff
@@ -194,190 +209,224 @@ Representation of a Neptune run in a limited read-only mode.
...
 ---
 
 #### `prefetch()`
+
 Pre-fetches a batch of fields to the internal cache.
 
-Improves the performance of access to consecutive field values. Only simple field types are supported (`int`, `float`, `str`).
+Improves the performance of access to consecutive field values.
+
+Supported Neptune field types: `Integer`, `Float`, `String`, `Boolean`, `Datetime`, and `StringSet`.
```
[the remark this answers is not in the document]
Comment (normandy7) [reply README.md:286]:
```suggestion
Supported Neptune field types: integer, float, string, Boolean, datetime, and state.
```

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (Raalsky) [reply README.md:304]: I've checked and looks like we have support for StringSets as well :wink:

[the remark this answers is not in the document]
Comment (normandy7) [reply README.md:413]: 👍🏻 I added a section on [line 543](https://github.com/neptune-ai/neptune-fetcher/pull/35/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R543)

**Review by Raalsky — APPROVED** [review]

**Review by AleksanderWWW — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -238,54 +261,67 @@ __Parameters__:
 | `ascending`       | `bool`, optional                              | `False`             | Whether to sort the entries in ascending order of the sorting column values.                                                                                                                                                                                                           |
 | `progress_bar`    | `bool`, `Type[ProgressBarCallback]`, optional | `None`              | Set to `False `to disable the download progress bar, or pass a type of ProgressBarCallback to [use your own progress bar](https://docs.neptune.ai/usage/querying_metadata/#using-a-custom-progress-bar). If set to `None` or `True`, the default tqdm-based progress bar will be used. |
 
-__Example__:
+__Returns:__ `pandas.DataFrame`: A pandas DataFrame containing metadata of the fetched experiments.
+
+> [!IMPORTANT]
+> When using a regular expression to filter experiments or columns, the total number of matched entries must not exceed 100.
+
+Specifically, you can fetch a data fram with a maximum of:
```
Comment (normandy7) [inline README.md:269]:
```suggestion
Specifically, you can fetch a data frame with a maximum of:
```

File: README.md [hunk 4]
```diff
@@ -172,45 +180,60 @@ __Returns__:
...
+| `ascending`       | `bool`, optional                              | `False`             | Whether to sort the entries in ascending order of the sorting column values. |
 | `progress_bar`    | `bool`, `Type[ProgressBarCallback]`, optional | `None`              | Set to `False `to disable the download progress bar, or pass a type of ProgressBarCallback to [use your own progress bar](https://docs.neptune.ai/usage/querying_metadata/#using-a-custom-progress-bar). If set to `None` or `True`, the default tqdm-based progress bar will be used. |
 
-__Example__:
+__Returns:__ `pandas.DataFrame`: A pandas DataFrame containing metadata of the fetched runs.
+
+> [!IMPORTANT]
+> When using a regular expression to filter runs or columns, the total number of matched entries must not exceed 100.
+
+Specifically, you can fetch a data fram with a maximum of:
```
Comment (normandy7) [inline README.md:206]:
```suggestion
Specifically, you can fetch a data frame with a maximum of:
```

**Comment by normandy7** [comment]
To be merged once `0.5.0` is released 🚀 @Raalsky

**Pull request merged**